### PR TITLE
523 proper info reporting

### PR DESCRIPTION
**Pull request by UlrichKu**
| Q             | A
| ------------- | ---
| Branch        | 6.3.x
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#523
| License       | MIT

### CODE REVIEW

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php [hunk 1]
```diff
@@ -144,7 +144,16 @@ public function ProcessIPNRequest($oOrder, $aURLParameter)
...
         if (0 != strcmp($sResponse, 'VERIFIED')) {
-            $logger->error("PayPal IPN: unable to send notify-validate response. Domain:{$sDomain}\nPath:{$sPath}\nParameter: ".print_r($aURLParameter, true)."\ndata: {$sData}\nRESPONSE: ".$sResponse);
+            $logger->error(
+                'PayPal IPN: unable to send notify-validate response.',
+                [
+                    'domain' => $sDomain,
+                    'path' => $sPath,
+                    'url_parameters' => $aURLParameter,
+                    'data' => $sData,
+                    'response' => $sResponse,
```
Comment (rnuernberg) [inline src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: please reference the order (id and ordernumber) - both are very helpful when searching for a problem with an ordern

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php [hunk 2]
```diff
@@ -156,7 +165,13 @@ public function ProcessIPNRequest($oOrder, $aURLParameter)
         if ($oPaymentParameterList->Length() > 0) {
             while ($oPaymentParameter = $oPaymentParameterList->Next()) {
                 if ('Completed' == $oPaymentParameter->fieldValue) {
-                    $logger->error("PayPal IPN: the txn '{$sTxnId}' has been processed before and was set to completed".print_r($aURLParameter, true).' data: '.$sData);
+                    $logger->info(
+                        "PayPal IPN: the txn '{$sTxnId}' has been processed before and was already set to completed.",
+                        [
+                            'url_parameters' => $aURLParameter,
+                            'data' => $sData,
```
Comment (rnuernberg) [inline src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: Please move sTxnId into the context and add order id and order number to the context

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php [hunk 3]
```diff
@@ -177,7 +192,13 @@ public function ProcessIPNRequest($oOrder, $aURLParameter)
 
         if (false === $bIsComplete) {
             // some other response other than is paid - we log the info
-            $logger->error("PayPal IPN: returned payment status '{$aURLParameter['payment_status']}'! ".$aURLParameter.' data: '.$sData);
+            $logger->error(
+                'PayPal IPN: returned payment status not recognized.',
+                [
+                    'url_parameters' => $aURLParameter,
+                    'data' => $sData,
```
Comment (rnuernberg) [inline src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: please add order id and ordernumber to the context

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php [hunk 4]
```diff
@@ -192,7 +213,12 @@ public function ProcessIPNRequest($oOrder, $aURLParameter)
 
         if (0 != strcasecmp($sCurrency, $sPaymentCurrency)) {
             // invalid currency
-            $logger->error('PayPal IPN: invalid currency in request '.print_r($aURLParameter, true));
+            $logger->error(
+                'PayPal IPN: invalid currency in request.',
+                [
+                    'url_parameters' => $aURLParameter
```
Comment (rnuernberg) [inline src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: please add order id and ordernumber to the context

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php [hunk 5]
```diff
@@ -219,7 +245,12 @@ public function ProcessIPNRequest($oOrder, $aURLParameter)
                 $bPaymentOk = true;
                 $bPaymentCompleted = true;
             } else {
-                $logger->error("PayPal IPN: invalid amount paid: required = {$oOrder->fieldValueTotal}, paid = {$dPaymentValue}. Rawdata: ".print_r($aURLParameter, true));
+                $logger->error(
+                    "PayPal IPN: invalid amount paid: required = {$oOrder->fieldValueTotal}, paid = {$dPaymentValue}.",
+                    [
+                        'url_parameters' => $aURLParameter
```
Comment (rnuernberg) [inline src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: Please move the variable parts of the message to the context and add the order id and order number to the context as well

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php [hunk 6]
```diff
@@ -240,7 +271,12 @@ public function ProcessIPNRequest($oOrder, $aURLParameter)
         }
 
         if (false == $bPaymentOk) {
-            $logger->error('PayPal IPN: payment invalid '.print_r($aURLParameter, true));
+            $logger->error(
+                'PayPal IPN: payment invalid.',
+                [
+                    'url_parameters' => $aURLParameter
```
Comment (rnuernberg) [inline src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: please add order id and ordernumber to the context

Comment (UlrichKu) [reply src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: ok

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php (hunk 5 above)
Comment (UlrichKu) [reply src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: ok

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php (hunk 4 above)
Comment (UlrichKu) [reply src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: ok

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php (hunk 3 above)
Comment (UlrichKu) [reply src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: ok

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php (hunk 2 above)
Comment (UlrichKu) [reply src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: ok

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php (hunk 1 above)
Comment (UlrichKu) [reply src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPal_PayViaLink.class.php]: ok

**Pull request merged**